op-e2e/actions/interop/emitter_contract_test.go

```diff
@@ -0,0 +1,214 @@
+	tx, err := inboxContract.ValidateMessage(auth, id, crypto.Keccak256Hash(payload))
+	require.NoError(t, err)
+
+	return tx
+}
+
+func idForTx(t helpers.Testing, actors *InteropActors, tx *types.Transaction) inbox.Identifier {
+	receipt, err := actors.ChainA.SequencerEngine.EthClient().TransactionReceipt(t.Ctx(), tx.Hash())
+	require.NoError(t, err)
+	block, err := actors.ChainA.SequencerEngine.EthClient().BlockByNumber(t.Ctx(), receipt.BlockNumber)
+	require.NoError(t, err)
+
+	return inbox.Identifier{
+		Origin:      *tx.To(),
+		BlockNumber: receipt.BlockNumber,
+		LogIndex:    common.Big0,
+		Timestamp:   big.NewInt(int64(block.Time())),
+		ChainId:     actors.ChainA.RollupCfg.L2ChainID,
+	}
+}
+
+func initializeChainState(t helpers.Testing, actors *InteropActors) {
+	// Initialize both chain states
+	actors.ChainA.Sequencer.ActL2PipelineFull(t)
+	actors.ChainB.Sequencer.ActL2PipelineFull(t)
+
+	// Sync supervisors
+	actors.ChainA.Sequencer.SyncSupervisor(t)
+	actors.ChainB.Sequencer.SyncSupervisor(t)
+
+	// Verify initial state
+	statusA := actors.ChainA.Sequencer.SyncStatus()
+	statusB := actors.ChainB.Sequencer.SyncStatus()
+	require.Equal(t, uint64(0), statusA.UnsafeL2.Number)
+	require.Equal(t, uint64(0), statusB.UnsafeL2.Number)
+
+	// Complete initial sync
+	actors.Supervisor.ProcessFull(t)
+}
+
+func initializeEmitterContractTest(t helpers.Testing, aliceA *userWithKeys, actors *InteropActors) *types.Transaction {
+	// Deploy message contract and emit a log on ChainA
+	auth := newL2TxOpts(t, aliceA.secret, actors.ChainA)
+	emitContract, tx, _, err := emit.DeployEmit(auth, actors.ChainA.SequencerEngine.EthClient())
+	require.NoError(t, err)
+	includeTxOnChain(t, actors, actors.ChainA, tx, aliceA.address)
+
+	emitTx := newEmitMessageTx(t, actors.ChainA, aliceA, emitContract, []byte("test message"))
+	includeTxOnChain(t, actors, actors.ChainA, emitTx, aliceA.address)
+	assertHeads(t, actors.ChainA, 2, 2, 2)
+
+	// Verify initial state
+	assertHeads(t, actors.ChainA, 2, 2, 2)
+	assertHeads(t, actors.ChainB, 0, 0, 0)
+
+	return emitTx
+}
+
+func includeTxOnChain(t helpers.Testing, actors *InteropActors, chain *Chain, tx *types.Transaction, sender common.Address) {
+	// Create L2 block on the given chain
+	chain.Sequencer.ActL2StartBlock(t)
+	if tx != nil {
+		err := chain.SequencerEngine.EngineApi.IncludeTx(tx, sender)
+		require.NoError(t, err)
+	}
+	chain.Sequencer.ActL2EndBlock(t)
+
+	// Sync the chain and the supervisor
+	chain.Sequencer.SyncSupervisor(t)
+	actors.Supervisor.ProcessFull(t)
+
+	// Add to L1
+	chain.Batcher.ActSubmitAll(t)
+	actors.L1Miner.ActL1StartBlock(12)(t)
+	actors.L1Miner.ActL1IncludeTx(chain.BatcherAddr)(t)
+	actors.L1Miner.ActL1EndBlock(t)
+
+	// Complete L1 data processing
+	chain.Sequencer.ActL2EventsUntil(t, event.Is[derive.ExhaustedL1Event], 100, false)
+	actors.Supervisor.SignalLatestL1(t)
+	chain.Sequencer.SyncSupervisor(t)
+	actors.Supervisor.SignalLatestL1(t)
```

Comment:
I think you need a `actors.Supervisor.ProcessFull(t)` here to make the cross-safe updating happen. The `SyncSupervisor()` only pulls events from the op-node into the supervisor, it doesn't complete the supervisor work. Although, maybe it's a good idea to make the supervisor do some immediate processing after pulling the events from the op-node.